cpp/src/c_api/k_core.cpp

```diff
@@ -119,7 +119,7 @@ struct k_core_functor : public cugraph::c_api::abstract_functor {
             do_expensive_check_);
       }
 
-      auto degree_type = reinterpret_cast<cugraph::k_core_degree_type_t>(degree_type);
+      cugraph::k_core_degree_type_t degree_type = reinterpret_cast<cugraph::k_core_degree_type_t>(degree_type);
```

Comment:
@moonwatcher Please fix this one, you might need to just cast  = (cugraph::k_core_degree_type_t)degree_type_; cause the compiler won't like it ...